addons/point_of_sale/tests/test_pos_basic_config.py

```diff
@@ -318,6 +318,68 @@ def test_orders_with_zero_valued_invoiced(self):
+            order = self.pos_session.order_ids.filtered(lambda order: '666-666-666' in order.pos_reference)
+
+            # refund
+            order.refund()
+            refund_order = self.pos_session.order_ids.filtered(lambda order: order.state == 'draft')
+
+            # pay the refund
+            context_make_payment = {"active_ids": [refund_order.id], "active_id": refund_order.id}
+            make_payment = self.env['pos.make.payment'].with_context(context_make_payment).create({
+                'payment_method_id': self.cash_pm1.id,
+                'amount': -100,
+            })
+            make_payment.check()
+
+            # invoice refund
+            refund_order.action_pos_order_invoice()
+
+        self._run_test({
+            'payment_methods': self.cash_pm1 | self.bank_pm1,
+            'orders': [
+                {'pos_order_lines_ui_args': [(self.product1, 10)], 'payments': [(self.cash_pm1, 100)], 'customer': self.customer, 'is_invoiced': True, 'uid': '666-666-666'},
+            ],
+            'before_closing_cb': _before_closing_cb,
+            'journal_entries_before_closing': {
+                '666-666-666': {
+                    'invoice': {
+                        'line_ids': [
+                            {'account_id': self.sales_account.id, 'partner_id': self.customer.id, 'debit': 0, 'credit': 100, 'reconciled': False},
+                            {'account_id': self.c1_receivable.id, 'partner_id': self.customer.id, 'debit': 100, 'credit': 0, 'reconciled': True},
+                        ]
+                    },
+                    'payments': [
+                        ((self.cash_pm1, 100), {
+                            'line_ids': [
+                                {'account_id': self.c1_receivable.id, 'partner_id': self.customer.id, 'debit': 0, 'credit': 100, 'reconciled': True},
+                                {'account_id': self.pos_receivable_account.id, 'partner_id': False, 'debit': 100, 'credit': 0, 'reconciled': False},
+                            ]
+                        }),
+                    ],
+                }
+            },
+            'journal_entries_after_closing': {
+                'session_journal_entry': {
+                    'line_ids': [
+                        {'account_id': self.pos_receivable_account.id, 'partner_id': False, 'debit': 0, 'credit': 0, 'reconciled': True},
+                    ],
+                },
+                'cash_statement': [
+                    ((100, ), {
+                        'line_ids': [
+                            {'account_id': self.cash_pm1.outstanding_account_id.id, 'partner_id': False, 'debit': 100, 'credit': 0, 'reconciled': False},
+                            {'account_id': self.cash_pm1.receivable_account_id.id, 'partner_id': False, 'debit': 0, 'credit': 100, 'reconciled': True},
+                        ]
+                    }),
+                ],
+                'bank_payments': [],
```

Comment:
not sure about this part, but test doesn't fail

Reply:
@caburj please, as you did the refactor of the pos tests (https://github.com/odoo/odoo/commit/c0f879f6b831a49f29317d3841d88dbb097b2ace), could you check if this part is correct? this test comes from https://github.com/odoo/odoo/pull/98498/commits/99030bdb17489bb8e18bf9bcc867f551368f77b7